### PR TITLE
Remove datagen config and fix argument marshaling

### PR DESCRIPTION
**Pull request by robertbastian**
This adds unnecessary complexity

### CODE REVIEW

File: provider/datagen/src/bin/icu4x-datagen.rs [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+        driver = driver.with_locales_no_fallback(
+            match preprocessed_locales {
+                Some(PreprocessedLocales::All) => {
+                    eyre::bail!("--without-fallback needs an explicit locale list")
+                }
+                Some(PreprocessedLocales::LanguageIdentifiers(lids)) => lids,
+                None => cli
+                    .locales
+                    .into_iter()
+                    .map(|family| family.parse().wrap_err(family))
```
Comment (sffc) [inline provider/datagen/src/bin/icu4x-datagen.rs:458]:
```suggestion
                    .map(|langid_str| langid_str.parse().wrap_err(langid_str))
```

File: provider/datagen/src/bin/icu4x-datagen.rs [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+        let locale_families = match preprocessed_locales {
+            Some(PreprocessedLocales::All) => vec![LocaleFamily::full()],
+            Some(PreprocessedLocales::LanguageIdentifiers(lids)) => lids
+                .into_iter()
+                .map(LocaleFamily::with_descendants)
+                .collect(),
+            None => cli
+                .locales
+                .into_iter()
+                .map(|family| family.parse().wrap_err(family))
```
Comment (sffc) [inline provider/datagen/src/bin/icu4x-datagen.rs:473]:
```suggestion
                .map(|family_str| family_str.parse().wrap_err(family_str))
```

File: provider/datagen/src/bin/icu4x-datagen.rs [hunk 3]
```diff
@@ -0,0 +1,628 @@
...
+            }
+            (Some(x), _, false) => match x {
+                DeduplicationStrategy::Maximal => Some(icu_datagen::DeduplicationStrategy::Maximal),
+                DeduplicationStrategy::None => Some(icu_datagen::DeduplicationStrategy::None),
+            },
+            (None, fallback_mode, false) => match fallback_mode {
+                Fallback::Auto => None,
+                Fallback::Hybrid => Some(icu_datagen::DeduplicationStrategy::None),
+                Fallback::Runtime => Some(icu_datagen::DeduplicationStrategy::None),
+                Fallback::RuntimeManual => Some(icu_datagen::DeduplicationStrategy::None),
```
Comment (sffc) [inline provider/datagen/src/bin/icu4x-datagen.rs:494]: Oops, I think this came from earlier
```suggestion
                Fallback::Runtime => Some(icu_datagen::DeduplicationStrategy::Maximal),
                Fallback::RuntimeManual => Some(icu_datagen::DeduplicationStrategy::Maximal),
```

**Review by sffc — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]

**Comment by robertbastian** [comment]
Can one of you try to merge this? The button just won't work for me

**Comment by sffc** [comment]
Hmmm, I get an error too

Merge attempt failed
Merge already in progress

**Pull request merged**